overlord/servicestate/quota_control.go

```diff
@@ -124,7 +134,14 @@ func CreateQuota(st *state.State, name string, createOpts CreateQuotaOptions) (*
 	}
 
 	// make sure the specified snaps exist and aren't currently in another group
-	if err := validateSnapForAddingToGroup(st, createOpts.Snaps, name, allGrps); err != nil {
+	parentGrp := allGrps[createOpts.ParentName]
+	if err := validateSnapForAddingToGroup(st, createOpts.Snaps, name, parentGrp, allGrps); err != nil {
+		return nil, err
+	}
+
+	// if services are provided, the make sure they refer to correct snaps and valid
+	// services.
+	if err := validateSnapServicesForAddingToGroup(st, createOpts.Services, name, parentGrp, allGrps); err != nil {
 		return nil, err
```

Comment:
there seem to be no test that fails here

Reply:
Added dedicated unit tests for this function in same package! Thanks!